var/spack/repos/builtin/packages/libxdmcp/package.py

```diff
@@ -17,4 +17,5 @@ class Libxdmcp(AutotoolsPackage):
     depends_on('xproto', type='build')
     depends_on('pkgconfig', type='build')
     depends_on('util-macros', type='build')
-    depends_on('libbsd')
+    depends_on('libbsd', when='platform=linux')
+    depends_on('libbsd', when='platform=cray')
```

Comment:
I recently added this dependency in #10122 because I couldn't get `libxdmcp` to build without it on Cray, but now it's causing problems on macOS.

Reply:
Yes, that's what I did.